pkg/models/kanban_task_bucket_test.go

```diff
@@ -157,3 +157,41 @@ func TestTaskBucket_Update(t *testing.T) {
 		}, false)
 	})
 }
+
+func TestTaskBucket_UpsertDuplicateError(t *testing.T) {
+	t.Run("test constraint violation detection", func(t *testing.T) {
+		db.LoadAndAssertFixtures(t)
+		s := db.NewSession()
+		defer s.Close()
+
+		// Try to insert a task that's already in one of the fixture buckets
+		// From fixtures: task_id: 1, project_view_id: 4, bucket_id: 1
+		tb := &TaskBucket{
+			TaskID:        1,
+			ProjectViewID: 4,
+			BucketID:      2, // Different bucket, but same task_id + project_view_id combination
+		}
+
+		// Directly insert to trigger constraint violation, bypassing the upsert logic
```

Comment:
Use the upsert logic! The goal here is to check if the duplicate key error is triggered correctly.